concordium-std/tests/state/map-multiple-state-ref-mut.rs

```diff
@@ -0,0 +1,17 @@
+//! This test checks that you cannot have multiple `StateRefMut` from a
+//! `StateMap` alive at the same time.
+//!
+//! When compiling it, the borrow-checker is supposed to throw an error.
+use concordium_std::*;
+
+pub fn main() {
+    let mut state_builder = StateBuilder::open(ExternStateApi::open());
+    let mut map: StateMap<u8, u8, _> = state_builder.new_map();
+    map.insert(0, 1);
+    map.insert(1, 2);
+    // Get two mutable references and unwrap the options.
+    let e1 = map.get_mut(&0u8).unwrap();
+    let e2 = map.get_mut(&1u8).unwrap();
```

Comment:
We don't really have to unwrap here.

Reply:
Well, if both of these are `None`, we have a bug

Reply:
nevermind, I realize the point of the test is to fail

Reply:
```suggestion
    let e1 = map.get_mut(&0u8);
    let e2 = map.get_mut(&1u8);
```